src/functions.rs

```diff
@@ -1931,14 +1932,27 @@ impl<'a> FunctionInfo<'a> {
         }
 
         let mut annotations = BTreeSet::new();
-        for class_alias in &deriving_type.get_aliases() {
+        for class_alias in deriving_type
+            .get_aliases()
+            .into_iter()
+            .chain(std::iter::once(&deriving_type.name))
+        {
             for func_alias in &derived_name_aliases {
                 // No alias for real class and func combo
-                if *class_alias == &deriving_type.name && name == func_alias {
+                if class_alias == &deriving_type.name && name == func_alias {
+                    continue;
+                }
+                // If we already define a function with this name, then we're overriding that
+                // alias, so we don't alias to it here. This includes functions we're in the
+                // process of deriving
+                if deriving_type.defines_function(func_alias.as_ref(), functions)
```

Comment:
Since both of these ifs lead to a `continue` we could combine them, but I get if you want to separate them for readability.

Reply:
Yeah, I think it's more readable to leave them separate.  It becomes quite the elaborate conditional if we combine, and this way it's natural to keep the comments separate as well for the two separate cases.